pkgs/sdk/server/src/FeatureFlagsState.cs

```diff
@@ -194,14 +209,15 @@ internal FeatureFlagsStateBuilder AddFlag(string flagKey, LdValue value, int? va
                 Reason = trackReason || (_withReasons && (!_detailsOnlyIfTracked || flagIsTracked)) ? reason : (EvaluationReason?)null,
                 DebugEventsUntilDate = flagDebugEventsUntilDate,
                 TrackEvents = flagTrackEvents,
-                TrackReason = trackReason
+                TrackReason = trackReason,
+                Prerequisites = prerequisites
             };
             _flags[flagKey] = flag;
             return this;
         }
     }
 
-    internal struct FlagState
+    internal struct FlagState : IEquatable<FlagState>
```

Comment:
Non-nullable prerequisites feel a bit strange to me.

Reply:
Fair. Is that from a "what a library consumer would expect in C#" kind of way, or a "how structs work" way?

Reply:
So, it feels strange in the language to me. We have null-state analysis, so we should be able to safely mark things null-able and use them. https://learn.microsoft.com/en-us/dotnet/csharp/nullable-references#null-state-analysis

From a struct perspective, ideally structs only contain other value types, but that isn't always possible. Next is for structs to only contain immutable reference types in addition to value types. Least favorable is this mix, but also not always avoidable. A struct with mostly not references, with value semantics, is better than a class here probably. But it doesn't behave like a struct. In that if you alias it and mutate the list, then both lists mutate.

Reply:
If we make an immutable list, and then assign the entire list, then it at least makes it more toward the normal side. It could also be exposed as an `IReadOnlyList<T>`.

Reply:
I don't think this is blocking though, as it isn't part of the external API.